File: backend/src/main/java/com/cristislv1/backend/workout/Workout.java
package com.cristislv1.backend.workout;

import jakarta.persistence.*;
import lombok.Getter;
import lombok.Setter;

import java.time.OffsetDateTime;
import java.util.UUID;
import java.util.ArrayList;
import java.util.List;

@Getter
@Setter
@Entity
@Table(name = "workouts")
public class Workout {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    @Column(name = "user_id", nullable = false)
    private UUID userId;

    @Column(nullable = false)
    private String type = "gym";

    @Column(name = "performed_at")
    private OffsetDateTime performedAt;

    @Column(name = "duration_seconds")
    private Integer durationSeconds;

    private String notes;

    @Column(name = "created_at")
    private OffsetDateTime createdAt;

    @OneToMany(mappedBy = "workout", cascade = CascadeType.ALL, orphanRemoval = true)
    private List<WorkoutExercise> exercises = new ArrayList<>();
    
    public void addExercise(WorkoutExercise exercise) {
        exercises.add(exercise);
        exercise.setWorkout(this);
    }
}

File: frontend/lib/models/workout.dart
import 'workout_exercise.dart';

class Workout {
  final int id;
  final String userId;
  final String type;
  final String performedAt;
  final int? durationSeconds;
  final String? notes;
  final String createdAt;
  final List<WorkoutExercise> exercises;

  Workout({
    required this.id,
    required this.userId,
    required this.type,
    required this.performedAt,
    this.durationSeconds,
    this.notes,
    required this.createdAt,
    required this.exercises,
  });

  factory Workout.fromJson(Map<String, dynamic> json) {
    var exercisesList = json['exercises'] as List? ?? [];
    List<WorkoutExercise> exercises = exercisesList.map((e) => WorkoutExercise.fromJson(e)).toList();

    return Workout(
      id: json['id'],
      userId: json['userId'],
      type: json['type'] ?? 'gym',
      performedAt: json['performedAt'],
      durationSeconds: json['durationSeconds'],
      notes: json['notes'],
      createdAt: json['createdAt'],
      exercises: exercises,
    );
  }
}
